src/main/java/io/getunleash/repository/SegmentCollection.java

```diff
@@ -0,0 +1,44 @@
+package io.getunleash.repository;
+
+import io.getunleash.Segment;
+import io.getunleash.lang.Nullable;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public final class SegmentCollection implements Serializable {
+
+    static final long serialVersionUID = 1214L;
+    private final Collection<Segment> segments;
+    private final transient Map<Integer, Segment> cache;
```

Comment:
nice. Are we sure this is correctly re-populated when segments are read from the backup file?

Reply:
I will write a test for it